plugin/rewrite/rewrite.go

```diff
@@ -42,30 +42,25 @@ func (rw Rewrite) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg
 	state := request.Request{W: w, Req: r}
 
 	for _, rule := range rw.Rules {
-		switch result := rule.Rewrite(ctx, state); result {
-		case RewriteDone:
+		result := rule.Rewrite(ctx, state)
+		if result == RewriteDone {
 			if !validName(state.Req.Question[0].Name) {
 				x := state.Req.Question[0].Name
 				log.Errorf("Invalid name after rewrite: %s", x)
 				state.Req.Question[0] = wr.originalQuestion
 				return dns.RcodeServerFailure, fmt.Errorf("invalid name after rewrite: %s", x)
 			}
 			respRule := rule.GetResponseRule()
-			if respRule.Active == true {
+			if !rw.noRevert && respRule.Active == true {
 				wr.ResponseRewrite = true
 				wr.ResponseRules = append(wr.ResponseRules, respRule)
 			}
 			if rule.Mode() == Stop {
-				if rw.noRevert {
-					return plugin.NextOrFailure(rw.Name(), rw.Next, ctx, w, r)
-				}
-				return plugin.NextOrFailure(rw.Name(), rw.Next, ctx, wr, r)
+				break
 			}
-		case RewriteIgnored:
```

Comment:
In the new code I can't see this case, being handled...

Also don't understand why the 'switch' was changed to a if-then-elses

Reply:
There was no code attached to `RewriteIgnore`.
I changed for switch to if, because it was possible and that allow to do a `break` in the loop.
Else I need a more complicated way to `break` out of the switch and the loop

Reply:
ok. I found the way with a labelled break. Will revert to initial switch.